Updated style guide (#564)
stackabletech/documentation: modules/contributor/pages/docs-style-guide.adoc

* https://kubernetes.io/docs/contribute/style/style-guide/#use-normal-style-for-string-and-integer-field-values[Use normal style for string and integer field values]
* https://kubernetes.io/docs/contribute/style/style-guide/#use-code-style-for-kubernetes-command-tool-and-component-names[Use `code style` for command line tools] such as `stackablectl`

=== Code blocks (scripts, console instructions)

For scripts and source code blocks, use this syntax:

[source]
....
[source,yaml] <.>
----
spec:
  command: | <.>
    some-command --with-long-option 'my-option' <.>
----
....
<.> Put the language here, i.e. `bash` or `yaml` to get appropriate syntax highlighting.

> <.> Indent 2 spaces to save horizontal space (also for Python or Java code!).

Reviewer: I didn't understand this bit. Is it talking about the YAML indentation (required anyway), or something else?

Author: maybe the yaml example is bad. This is about not using 4 spaces (as is common in many other languages), to save the horizontal space on the website. Otherwise eventually you have to scroll sideways to see the code block.

Does this already help? https://github.com/stackabletech/documentation/pull/564/commits/8235b513b8f58bae7e173b12572a9b44db2f08aa

Reviewer: > Indent 2 spaces (not 4, also for Python/Java/Rust etc.) to save horizontal space.

Understood. I think it can make sense, but also becomes a bit of a pain when pasting larger sections of code that doesn't use 2-space indentations.

Maybe it can be a recommendation when you need to save horizontal space?

<.> Use long options for shell script flags, as they are easier to understand. Use your own judgment; common ones like `rm -rf` can still be short flags.

> For console instructions, use:

Reviewer: ```suggestion
For console instructions, which might include example output:
```

[source]
....
[source,console] <.>
----
$ echo 'Hello World' <.>
Hello World <.>
----
....
<.> Use `console` for the highlighting setting.

> <.> Prefix the command line with the dollar sign (`$`).

Reviewer: The reason being that the copy functionality will copy the command line (without the `$`, which is also not selectable), and not the output below it.

Not sure how that could be worded to fit nicely. or maybe it can go below?

Author: https://github.com/stackabletech/documentation/pull/564/commits/8235b513b8f58bae7e173b12572a9b44db2f08aa I added some explanation, what do you think?

Reviewer: Understood. I think it can make sense, but also becomes a bit of a pain when pasting larger sections of code that doesn't use 2-space indentations.

Maybe it can be a recommendation when you need to save horizontal space?

----
....
<.> Put the language here, i.e. `bash` or `yaml` to get appropriate syntax highlighting.

> <.> Indent 2 spaces to save horizontal space when applicable.

Reviewer: :raised_hands:

----
....
<.> Use `console` for the highlighting setting.

> <.> Prefix the command line with the dollar sign (`$`) which will copy the line in the 'Copy' functionality in the UI.

Reviewer: ```suggestion
<.> Prefix the command line with the dollar sign (`$`) so that when the reader clicks the 'Copy' button, only the command-lines are copied.
```

> <.> Do _not_ prefix output lines, so they are not copied.

Reviewer: ```suggestion
<.> Do _not_ prefix output lines, to prevent the lines from being copied.
```
